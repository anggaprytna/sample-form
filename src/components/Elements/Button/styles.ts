import {StyleSheet} from 'react-native';
import {colors, sizes, fonts} from '../../../theme/index';

const styles = StyleSheet.create({
  primary: {
    backgroundColor: colors.primary2,
    paddingVertical: 15,
    borderRadius: sizes.radiusMedium,
    alignItems: 'center',
  },
  dark: {
    backgroundColor: colors.dark,
    paddingVertical: 20,
    borderRadius: 15,
    alignItems: 'center',
  },
  light: {
    backgroundColor: colors.light,
    paddingVertical: 20,
    borderRadius: 15,
    alignItems: 'center',
  },
  textLight: {
    color: colors.black2,
    fontWeight: 'bold',
  },
  textDark: {
    color: colors.white,
    fontWeight: 'bold',
  },
  textPrimary: {
    color: colors.white,
    fontFamily: 'Poppins-SemiBold',
    fontSize: 16,
  },
});

export default styles;
