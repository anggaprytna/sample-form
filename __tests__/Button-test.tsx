/**
 * @format
 */

import 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import Button from '../src/components/Elements/Button';

it('Button', () => {
  // test props
  let buttonPrimary = renderer.create(<Button title="asd" type="primary" />);
  let buttonDark = renderer.create(<Button title="asd" type="dark" />);
  let buttonLight = renderer.create(<Button title="asd" type="light" />);
  expect({buttonPrimary, buttonDark, buttonLight}).toMatchSnapshot();
});
