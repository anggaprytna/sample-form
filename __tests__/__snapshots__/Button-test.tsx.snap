// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button 1`] = `
Object {
  "buttonDark": <View
    accessible={true}
    focusable={false}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#000",
          "borderRadius": 15,
          "marginHorizontal": 20,
          "paddingVertical": 20,
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFF",
              "fontWeight": "bold",
            },
            false,
            false,
          ]
        }
      >
        asd
      </Text>
    </View>
  </View>,
  "buttonLight": <View
    accessible={true}
    focusable={false}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#f4f4f4",
          "borderRadius": 15,
          "marginHorizontal": 20,
          "paddingVertical": 20,
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
        }
      }
    >
      <Text
        style={
          Array [
            false,
            false,
            Object {
              "color": "#4B4F71",
              "fontWeight": "bold",
            },
          ]
        }
      >
        asd
      </Text>
    </View>
  </View>,
  "buttonPrimary": <View
    accessible={true}
    focusable={false}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#003F6D",
          "borderRadius": 6,
          "marginHorizontal": 20,
          "paddingVertical": 15,
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
        }
      }
    >
      <Text
        style={
          Array [
            false,
            Object {
              "color": "#FFF",
              "fontWeight": "bold",
            },
            false,
          ]
        }
      >
        asd
      </Text>
    </View>
  </View>,
}
`;
